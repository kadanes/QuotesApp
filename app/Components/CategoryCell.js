import React, {Component} from 'react'
import { TouchableOpacity, View, Text, StyleSheet } from 'react-native'

export default class CategoryCell extends Component {
    render() {
        return (
            <TouchableOpacity 
                onPress={() => this.props.navigation.navigate('Quotes',{ id: this.props.id})}
            >
                <View style={styles.Container}>
                    <View style={styles.CategoryHeader}>
                        <Text style={styles.Category}> {this.props.Category.category} </Text>
                        <Text style={styles.QuotesCount}>{this.props.Category.quotes.length} </Text>
                    </View>
                    <View style={styles.AuthorBlock}>
                        {this.props.Category.quotes.map(item => <Text style={styles.Author}> {item.person}</Text>)}
                    </View>
                </View>
            </TouchableOpacity>
        )
    }
}


const styles = StyleSheet.create({
    Container: {
        margin: 10,
        padding: 10 ,
        backgroundColor: '#ff6347',
        borderRadius: 15,
        flex: 1
    },
    CategoryHeader: {
        flexDirection: 'row',
        flex: 1,
        justifyContent: 'space-between',
        
    },
    AuthorBlock: {
        flex: 1,
        flexDirection: 'column',
        marginTop: 10,
        marginLeft: 5
    },
    Category: {
        textTransform: 'capitalize',
        fontSize: 20,
        color: 'white',
        fontWeight: 'bold'
    },
    QuotesCount: {
        color: 'white',
        fontWeight: '600',
        fontSize: 12
    },
    Author: {
        color: '#E0E0E0',
        fontSize: 14
    }
})
