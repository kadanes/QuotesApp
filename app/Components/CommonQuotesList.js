import React, { Component} from 'react'
import { FlatList, View, Text } from 'react-native'
import { connect } from 'react-redux'

class CommonQuotesList extends Component {

    render () {
        return (
            <View style={styles.Container}>
                    <Text>{parentId}</Text>
                    <FlatList
                    style= {{flex:1}}
                    data= {quotes}
                    renderItem = {({item,index}) => {
                        console.log("Row: ",item.text,this.props.favourites.includes(`${parentId}_${index}`))
                        return (
                            <View >
                                
                                <Text>{item.text}</Text>
                                <Text>-- {item.person}</Text>
                                {  
                                    this.props.favourites.includes(`${parentId}_${index}`) ?
                                    <Button onPress={() => this.props.unsave(parentId,index)} title={'UnFavourite'}/ >:
                                    <Button onPress={() => this.props.save(parentId,index)} title={'Favourite'}/ > 
                                }
                                
                            </View>
                        )
                    }}
                    keyIterator = {(item,index) => index}
                    />  
            </View>
        )
    }
}