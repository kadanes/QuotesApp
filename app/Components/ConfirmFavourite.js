import React, {Component} from 'react'
import { Button, View, Text, StyleSheet } from 'react-native'

export default class ConfirmFavourite extends Component {

    render() {
        return (
            <View style={styles.modalContainer}>
                <Text>
                    Save quote? 
                </Text>
                <Button
                    title = 'Close'
                    onPress = {() => this.props.navigation.navigate('Favourites')} 
                />
            </View>
        )
       
    }

}

const styles = StyleSheet.create({
    modalContainer: {
        height: 300,
        width: 300,
        borderWidth: 2,
        borderColor: "red",
        justifyContent: 'center',
        alignItems: 'center',
    }
})