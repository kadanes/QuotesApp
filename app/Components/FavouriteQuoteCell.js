import React from 'react'
import { View, Text, TouchableHighlight, StyleSheet } from 'react-native'

const FavouriteQuoteCell = (props) => {
    return (
        <TouchableHighlight   
            onPress={() => props.navigation.navigate('Quotes',{id: props.item.parentId, category: props.item.category})}
            style={styles.TableCell}
        >
        <View>
            <Text style={styles.Quote}>{props.item.text}</Text>
            <Text style={styles.Author}>-- {props.item.person}</Text>
            <View style={styles.CategoryPill}>
                <Text style={styles.Category}>
                    {props.item.category}
                </Text>
            </View> 
        </View>
        
        </TouchableHighlight> 
    )
}

export default FavouriteQuoteCell

const styles = StyleSheet.create({
    TableCell: {
        backgroundColor: '#ff6347',
        margin:5,
        padding: 5,
        justifyContent: 'space-around',
        flexDirection: 'column',
        flex: 1 ,
        padding: 10,
        margin: 5,
        borderRadius: 15, 
    },
    "Quote": {
        fontWeight: 'bold',
        color: 'white'
    },
    "Author": {
        fontWeight:'200',
        color:'white',
        justifyContent: 'flex-end',
        alignItems: 'flex-end',
        height: 20
    }, 
    Category: {
        color: '#ff6347',
        fontWeight: 'bold',
        fontSize: 12,
        textTransform: 'capitalize',
        margin: 'auto'
    },
    CategoryPill: {
        marginTop: 10,
        padding: 2,
        height: 20,
        borderRadius: 10,
        backgroundColor: 'white',
        width: 100, 
        flex: 1,
        justifyContent: 'center',
        alignItems: 'center'    
    }
})