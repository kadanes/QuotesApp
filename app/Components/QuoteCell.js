import React,{Component} from 'react'
import {StyleSheet, View, Text, TouchableOpacity } from 'react-native'
import Ionicons from 'react-native-vector-icons/Ionicons'
import { connect } from 'react-redux'
import { watchSaveQuote, watchUnsaveQuote } from '../actions/Quotes'

export class QuoteCell extends Component {
    render() {

        return (
            <View style={styles.TableCell}  key={this.props.index}>
                <View style={{flex:1,flexDirection:'row', alignItems: 'center'}}>
                    <View style={{flex:1,flexDirection:'column', justifyContent: 'space-between'}}>
                        <Text style={styles.Quote}>{this.props.quote.text}</Text>
                        <Text style={styles.Author}>-- {this.props.quote.person}</Text>
                    </View>
                    
                    {  
                        this.props.favourites.includes(`${this.props.parentId}_${this.props.index}`) ?
                        <TouchableOpacity onPress={() => {
                            this.props.unsave(this.props.parentId,this.props.index)
                        }}>  
                            <Ionicons name='ios-remove-circle' style={styles.Button}/> 
                        </TouchableOpacity>:
                        <TouchableOpacity  onPress={() => {
                            
                            this.props.save(this.props.parentId,this.props.index)
                        }}> 
                            <Ionicons name='ios-add-circle' style={styles.Button}/>
                        </TouchableOpacity> 
                    }
                
                </View>
            </View>

        )
            
    }
}

const matchStateToProps = (state) =>{ 
    console.log("Quote list: ",state)
    return {
        favourites: state.favourites
    }
}

const matchDisptachToProps = (dispatch) => ({
    save : (parentId, id) => {
        dispatch(watchSaveQuote(parentId,id))
    },
    unsave : (parentId, id) => {
        dispatch(watchUnsaveQuote(parentId,id))
    }

})

export default connect(matchStateToProps,matchDisptachToProps)(QuoteCell)


const styles = StyleSheet.create({
    Container: {
        flex: 1,
        justifyContent: 'center',
        alignItems: 'center',
    },
    TableCell: {
        backgroundColor: '#ff6347',
        margin:5,
        padding: 20,
        borderRadius: 15
     
    },
    "Quote": {
        fontWeight: 'bold',
        color: 'white'
    },
    "Author": {
        fontWeight:'200',
        color:'white',
        height: 20, 
        marginTop: 10  
    },
    "Button": {
        color: "#DCDCDC",
        fontWeight:'700',
        fontSize: 30
    }
})