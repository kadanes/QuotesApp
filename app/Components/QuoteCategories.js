import React, {Component} from 'react'
import {TouchableOpacity, Button, Text, View, FlatList, StyleSheet, ActivityIndicator} from 'react-native'
import CategoryCell from '../Components/CategoryCell'
import { connect } from 'react-redux'


class QuotesCategories extends Component {

    static navigationOptions = {
        title: 'Categories',
        headerRight: (
            <Button
              onPress={() => alert('This is a button!')}
              title="Info"
              color="#fff"
            />
          )
    }

    render() {

        if (this.props.error == true) {
            return(
                <View style={styles.Container}>
                    <Text style={{color: 'red'}}>FAILED TO LOAD DATA</Text>
                    <Button 
                        title='Reload'
                        onPress={this.makeRemoteRequest}
                    />
                </View>
            )
        } else if (this.props.loading) {
            return(
                <View style={styles.Container}>
                    <ActivityIndicator size="large"/>
                </View>
            )
        } else if (!this.props.loading && !this.props.error) {
            return (
                <View style={styles.Container}>
                    <FlatList
                    style= {{flex:1, width: '100%'}}
                    data= {this.props.data}
                    renderItem = {({item,index}) => {
                        return (
                            
                            <TouchableOpacity 
                                onPress={() => this.props.navigation.navigate('Quotes',{quotes: item.quotes, id: index, category: item.category})}
                            >
                         
                                <CategoryCell Category={item}/>
          
                            </TouchableOpacity>  
                        )
                    }}
                    keyIterator = {(item,index) => index}
                    />  
                
                </View>
                   
            )
        }
            
    }    
}

const styles = StyleSheet.create({
    Container: {
        flex: 1,
        justifyContent: 'center',
        alignItems: 'center',
    }
})

const mapStateToProps = (state,props) => {
    return {
        loading: state.loading,
        error: state.error,
        data: state.data
    }
}
export default connect(mapStateToProps)(QuotesCategories)
