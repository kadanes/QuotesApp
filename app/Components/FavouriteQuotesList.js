import React, {Component} from 'react'
import {View, Text, FlatList, StyleSheet, ActivityIndicator, TouchableHighlight} from 'react-native'
import { connect } from 'react-redux'
import { SafeAreaView } from 'react-navigation'
import FavouriteQuoteCell from './FavouriteQuoteCell'
import {getFavouriteQuotes} from '../Reducer/Quotes'

export class FavouriteQuotesList extends Component {
    
    static navigationOptions = {
        title: 'Favourites'
    }

    render() {
        if (this.props.loading) {
            return(
                <View style={styles.Container}>
                    <ActivityIndicator size="large"/>
                </View> 
            )
        } else if (!this.props.loading) {
            return (
                <SafeAreaView style={styles.Container}>
                    <View style={{width: '100%', height: '100%'}}>
                        <FlatList
                        style= {{flex:1, width: '100%'}}
                        data= {this.props.favouritesCombi}
                        renderItem = {({item}) => this.renderItem(item,this.props.navigation)}
                        keyIterator = {(item,index) => index}
                        />  
                    </View>
                </SafeAreaView>       
            )
        }
    }

    renderItem(item,navigation) {
        return (
            <FavouriteQuoteCell 
                navigation={navigation} 
                item={item}
            />
        )
    }    
}

const mapStateToProps = (state) => ({
        loading: state.favouritesLoading,
        favouritesCombi: getFavouriteQuotes(state)
    })


const styles = StyleSheet.create({
    Container: {
        flex: 1,
        justifyContent: 'center',
        alignItems: 'center',
    }
})

export default connect(mapStateToProps)(FavouriteQuotesList)

