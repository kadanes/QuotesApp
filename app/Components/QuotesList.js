import React, {Component} from 'react'
import { View, Text, StyleSheet, FlatList, TouchableOpacity } from 'react-native' 
import { connect } from 'react-redux'
import { watchSaveQuote, watchUnsaveQuote } from '../actions/Quotes'
import QuoteCell from '../Components/QuoteCell'

export class QuotesList extends Component {

    static navigationOptions = ({ navigation }) => {
        category = navigation.getParam('category', 'Quotes')
        return {
        title: category.charAt(0).toUpperCase() + category.slice(1),
        }
    };

    render() {
       
        const { navigation } = this.props
        const parentId = navigation.getParam('id','nil')

        const quotes = this.props.quotes[parentId].quotes

        return (
            <View style={styles.Container}>
                    <FlatList 
                    style= {{flex:1, width: '100%'}}
                    data= {quotes}
                    renderItem = {({item,index}) => 
                         <QuoteCell
                               quote={item}
                               parentId={parentId}
                               index={index}     
                        />
                    }
                    keyIterator = {(_,index) => index}
                    />  
            </View>
        )
    }
}

const styles = StyleSheet.create({
    Container: {
        flex: 1,
        justifyContent: 'center',
        alignItems: 'center',
    }
})

const matchStateToProps = (state) => ({ 

        favourites: state.favourites,
        quotes: state.data
    
})

export default connect(matchStateToProps)(QuotesList)