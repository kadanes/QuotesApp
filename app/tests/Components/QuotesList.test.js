import React from 'react'
import {shallow} from 'enzyme'

import {QuotesList} from '../../Components/QuotesList'
import quotes from '../fixtures/quotes-fixture'
import favourites from '../fixtures/favourites-fixture'

const getParam = (id, def = 'null') => {
    switch (id) {
        default:
        return 0
        case 'category':
        return 'Inspirational'
    }
}

describe('Testing Quote List', () => {

    let wrapper,navigation

    beforeEach(()=>{
        
        let navigate = jest.fn()
        navigation = {
            navigate,
            getParam
        }

        wrapper = shallow(
                <QuotesList 
                    favourites={favourites}    
                    quotes={quotes}
                    navigation={navigation}
                />
            )
    })

    it('should render QuoteList',() => {
        expect(wrapper).toMatchSnapshot()
    })
})