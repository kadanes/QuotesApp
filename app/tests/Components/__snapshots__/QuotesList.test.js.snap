// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Quote List should render QuoteList 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <FlatList
    data={
      Array [
        Object {
          "id": "1",
          "person": "Theodore Roosevelt",
          "text": "Believe you can and you\\"re halfway there",
        },
        Object {
          "id": "2",
          "person": "Norman Vincent Peale",
          "text": "Change your thoughts and you change your world.",
        },
        Object {
          "id": "3",
          "person": "Robert H. Schuller",
          "text": "What great thing would you attempt if you knew you could not fail?",
        },
      ]
    }
    disableVirtualization={false}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    keyIterator={[Function]}
    maxToRenderPerBatch={10}
    numColumns={1}
    onEndReachedThreshold={2}
    renderItem={[Function]}
    scrollEventThrottle={50}
    style={
      Object {
        "flex": 1,
        "width": "100%",
      }
    }
    updateCellsBatchingPeriod={50}
    windowSize={21}
  />
</View>
`;
